### PR TITLE
Extra reference docs support

### PR DESCRIPTION
**Pull request by glicht**
## Status
Ready

## Related Issues
related: https://github.com/demisto/etc/issues/23431

## Description
Support for extra content docs which are not part of the content repo such as native integrations and external integrations. Add some docs:
* syslog
* remote access
* saml 2.0 - okta

## Screenshots
Paste here any images that will help the reviewer

### CODE REVIEW

File: .github/config.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  Hello World! Welcome to our Content Developer Docs project!
+  Thanks and congrats for opening your very first issue.
+  If you know the fix, you may also submit a PR to fix the issue if you like! 
+   
+newPRWelcomeComment: |
+  Welcome! Thank you and congrats on your first pull request.
+  
+  **What happens next?**
+  * Netlify is building a preview of the site with your modifications.
+  * Once the Netlify build is complete a preview will be available  at the _Checks_ section of the PR under _deploy/netlify_.
```
Comment (Itay4) [inline .github/config.yml:11]:
```suggestion
  * Once the Netlify build is complete a preview will be available at the _Checks_ section of the PR under _deploy/netlify_.
```

File: mdx-parse-server.js [hunk 2]
```diff
@@ -1,12 +1,6 @@
 const mdx = require('@mdx-js/mdx');
 const http = require('http')
 
-async function parseMDX(file) {
```
Comment (Itay4) [inline mdx-parse-server.js:4]: this was not used?

File: content-repo/gendocs.py [hunk 3]
```diff
@@ -220,12 +220,54 @@ def index_doc_infos(doc_infos: List[DocInfo], link_prefix: str, headers: Optiona
...
+        verify_mdx_server(content)
+        with open(f'{target_dir}/{file_id}.md', mode='w', encoding='utf-8') as f:
+            f.write(content)
+        return DocInfo(file_id, name, desc, readme_file)
+    except Exception as ex:
+        print(f'fail: {readme_file}. Exception: {traceback.format_exc()}')
+        return DocInfo('', '', '', readme_file, str(ex).splitlines()[0])
+
+
+def process_extra_docs(target_dir: str, prefix: str):
```
Comment (Itay4) [inline content-repo/gendocs.py:248]:
```suggestion
def process_extra_docs(target_dir: str, prefix: str) -> Generator[DocInfo]:
```

File: mdx-parse-server.js (hunk 2 above)
Comment (glicht) [reply mdx-parse-server.js:4]: No. It is old code copied from another file.

**Comment by glicht** [comment]
@kirbles19 please review:

* https://github.com/demisto/content-docs/edit/extra-ref-docs/content-repo/extra-docs/integrations/remote-access.md?pr=%2Fdemisto%2Fcontent-docs%2Fpull%2F113
* https://github.com/demisto/content-docs/edit/extra-ref-docs/content-repo/extra-docs/integrations/syslog.md?pr=%2Fdemisto%2Fcontent-docs%2Fpull%2F113
* https://github.com/demisto/content-docs/edit/extra-ref-docs/content-repo/extra-docs/integrations/saml-okta.md?pr=%2Fdemisto%2Fcontent-docs%2Fpull%2F113

These have been migrated from zendesk

**Comment by kirbles19** [comment]
These two look like they could use some work
- https://github.com/demisto/content-docs/edit/extra-ref-docs/content-repo/extra-docs/integrations/remote-access.md?pr=%2Fdemisto%2Fcontent-docs%2Fpull%2F113
- https://github.com/demisto/content-docs/edit/extra-ref-docs/content-repo/extra-docs/integrations/syslog.md?pr=%2Fdemisto%2Fcontent-docs%2Fpull%2F113

This one is in tech docs
- https://github.com/demisto/content-docs/edit/extra-ref-docs/content-repo/extra-docs/integrations/saml-okta.md?pr=%2Fdemisto%2Fcontent-docs%2Fpull%2F113

**Pull request merged**